core/dataset.cc

```diff
@@ -551,8 +551,12 @@ queryStructured(const SelectExpression & select,
     std::mutex lock;
     std::vector<MatrixNamedRow> output;
 
+    if (!having.isConstantTrue() && groupBy.clauses.empty())
+        throw HttpReturnException(400, "HAVING expression requires a GROUP BY expression");
```

Comment:
According to the "internet" a having absolutely requires a group by. If you have aggregators but no group by, you get only 1 row anyway, so filtering that row has little use. Nic can overrule me here though...